plugin/kubernetes/controller.go

```diff
@@ -365,21 +365,21 @@ func (dns *dnsControl) PodIndex(ip string) (pods []*api.Pod) {
 	return pods
 }
 
-func (dns *dnsControl) SvcIndex(idx string) (svcs []*api.Service) {
+func (dns *dnsControl) SvcIndex(key string) (svcs []*api.Service) {
 	if dns.svcLister == nil {
 		return nil
 	}
-	os, err := dns.svcLister.ByIndex(svcNameNamespaceIndex, idx)
+	o, _, err := dns.svcLister.GetByKey(key)
 	if err != nil {
 		return nil
 	}
-	for _, o := range os {
-		s, ok := o.(*api.Service)
-		if !ok {
-			continue
-		}
-		svcs = append(svcs, s)
+	s, ok := o.(*api.Service)
+
+	if !ok {
+		return
 	}
+	svcs = append(svcs, s)
```

Comment:
this doesn't need append.
I've we decide to merge #2128 it has a whole slew of fixes here.

Reply:
yeah - thanks for the reminder. I intend to go in and change these once we decide this PR is a viable route ... It means changing the function signature, and its an interface function, so it's a lot of changes.